src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -189,7 +189,12 @@ RadixContainer<T> materialize_input(const std::shared_ptr<const Table>& in_table
 /*
 Build all the hash tables for the partitions of Left. We parallelize this process for all partitions of Left
 */
-template <typename LeftType, typename HashedType>
+
+// For semi and anti joins, we only care about existence, so there is no point in adding a second occurence
```

Comment:
```c++
// For semi and anti joins, we only care whether a value exists or not, so there is no point in tracking more than one occurrence of a value.
// That is, znless we have secondary predicates: In those cases, we need to track all occurrences of a value 
enum class JoinHashBuildMode : { MultiMap, UniqueMap };

Reply:
My bike-shedding comes from having previously dissected the JoinHash operator: It's a hard thing to do. I'd rather make it as approachable as possible for the future.

Reply:
```
// For semi and anti joins, we only care whether a value exists or not, so there is no point in tracking the position
// in the input table of more than one occurrence of a value. However, if we have secondary predicates, we do need to
// track all occurrences of a value as that first position might be disqualified later.
enum class JoinHashBuildMode : bool { AllPositions, SinglePosition };
```

Reply:
this I like. still do not know why the bool, but that's not blocking this.

Reply:
Will remove it. Might have copy pasted it...